filewatcher/filewatcher.go

```diff
@@ -161,14 +227,21 @@ func (r *Result) watcherLoop(
 				continue
 			}
 
-			if filepath.Base(ev.Name) != file {
-				continue
-			}
-
 			switch ev.Op {
 			default:
 				// Ignore uninterested events.
 			case fsnotify.Create, fsnotify.Write:
+				// If a directory is created in a directory watcher we want
+				// to watch that as well
+				isDir, err := isDirectory(path)
+				if err != nil {
+					logger.Log(context.Background(), "filewatcher: isDirectory error: "+err.Error())
+					continue
+				}
+				if isDir {
+					refreshWatched()
+				}
```

Comment:
this should be moved into `parse` instead, before calling the parser callback (so we only refresh it once per parser call, not for every fsnotify event)